### DISCUSSION TITLE
Pycharm IDE doesn't recognize methods in InteractionResponse

### DISCUSSION THREAD

**Discussion by SeoulSKY** [Rapptz/discord.py#9504] [Q&A]
It shows a warning message and code completion doesn't work. However, the code runs with no problem. How to fix this issue?

<img width="785" alt="Screenshot 2023-07-21 at 12 41 13 PM" src="https://github.com/Rapptz/discord.py/assets/48105703/7729ec4b-f51d-43f9-bd0e-f9b2061a340f">

**Comment by LostLuma** [comment]
As far as I know there's no way around this at the moment as Pycharm's type checker can not yet handle quite a few typing features used by discord.py and other modern Python projects.

If you'd like to make use of static type checking you can use the pyright/pylance CLI or switch to VS Code to make use of these in editor (make sure to set `Python > Analysis > Type Checking Mode` to `basic` instead of `off` in your settings after installing the extension).

**Comment by TimG233** [comment]
Some alternative "solution": https://stackoverflow.com/questions/76577197/no-context-hint-and-false-unresolved-reference-on-discord-interaction-response <br>
As LostLuma said, there is no way to solve this directly in Pycharm. But you could specifically close these types of checking scenarios if you wish (in the link above).

**Reply by Soheab** [reply]
Another "solution" can be the following IIRC:
```py
response: discord.InteractionResponse = interaction.response  # type: ignore
...
await response.send_message(...)
...
```

**Reply by Beengoo** [reply]
or:

```py
await interaction.response.send_message(...)  # type: ignore
```
the same but shorter

*But i think better to use `interaction.response.defer()` and `interaction.followup.send()` you can avoid errors like `404 Unknown interaction`*

**Reply by CaedenPH** [reply]
> or:
> 
> 
> 
> ```py
> 
> await interaction.response.send_message(...)  # type: ignore
> 
> ```
> 
> the same but shorter
> 
> 
> 
> *But i think better to use `interaction.response.defer()` and `interaction.followup.send()` you can avoid errors like `404 Unknown interaction`*

Using `type: ignore` to disable the line is a hacky fix which shouldn't be used on a correctly typed line
